### DISCUSSION TITLE
Android: App non-responsive while SWF Loading

### DISCUSSION THREAD

**Discussion by gaurangtalwadkar** [airsdk/Adobe-Runtime-Support#641] [Q&A]
**Environment:**
AIR SDK:  29
Android API: 15+

**Problem Description:**
We are using Loader class for loading our game swf file which is above 30MB.
But while loading the SWF (even though load function is non-blocking), touch or device back button doesn't work. It is making app non-responsive during that period. We are using below code.
```
var swfLoader = new Loader();
var urlReq:URLRequest = new URLRequest("SWF_PATH");
var lc:LoaderContext = new LoaderContext(true, ApplicationDomain.currentDomain);
swfLoader.load(urlReq, lc);
```
Please look into this issue. Is there any way to load an SWF file without making app non-responsive?

**Comment by ajwfrost** [comment]
Hi - can you confirm whether the large SWF is found within your APK package, or is this being downloaded from a server? 

You're right that it should be non-blocking, and it's definitely asynchronous, so we'll check the thread priorities and see how the loading/downloading is actually implemented..

thanks

**Comment by pkumargsngames** [comment]
The SWF is part of APK.

Additional note: we also tried attaching a click handler using
this.addEventListener(MouseEvent.CLICK, onClick1);
private function onClick1(evt:MouseEvent):void
{
    // code to log timestamp and some test message
}

This click handler works till we call the load method. Once SWF is loaded, then also it works. During load, it does not work.
